src/dataLoader/savedItemsDataLoader.ts

```diff
@@ -41,10 +41,54 @@ export async function batchGetSavedItemsByUrls(
 }
 
 /**
- * Creates a dataloader for saved items
+ * Reorders savedItems based on the id that are received to
+ * keep the response in the same order of savedItemIds as requested
+ * by the Apollo Gateway (Client API)
+ * @param ids
+ * @param savedItems
+ */
+export function reorderSavedItemsByIds(ids: string[], savedItems: SavedItem[]) {
+  const idToSavedItemMap = savedItems.reduce((acc, savedItem) => {
+    return {
+      ...acc,
```

Comment:
Don't need to spread this, can just rely on mutability and assign the key to the accumulator. If fetching many IDs it will be much more efficient

Reply:
done